buildtools.bzl

```diff
@@ -0,0 +1,152 @@
+        arch = asset.arch
+    if name == None or platform == None or arch == None:
+        fail("An asset or name/platform/arch must be specified.")
+
+    return "{name}_{platform}_{arch}".format(
+        name = name,
+        platform = platform,
+        arch = arch,
+    )
+
+def _asset_to_json(asset):
+    """Returns the JSON representation for an asset `struct` or a `list` of asset `struct` values.
+
+    Args:
+        asset: An asset `struct` as returned by `buildtools.create_asset`.
+
+    Returns:
+        Returns a JSON `string` representation of the provided value.
+    """
+    return json.encode(asset)
+
+def _asset_from_json(json_str):
+    """Returns an asset `struct` or a `list` of asset `struct` values as represented by the JSON `string`.
+
+    Args:
+        json_str: A JSON `string` representing an asset or a list of assets.
+
+    Returns:
+        An asset `struct` or a `list` of asset `struct` values.
+    """
+    result = json.decode(json_str)
+    if types.is_list(result):
+        return [_create_asset(**a) for a in result]
+    elif types.is_dict(result):
+        return _create_asset(**result)
+    fail("Unexpected result type decoding JSON string. %s" % (json_str))
+
+def _create_assets(version, names = _TOOL_NAMES, platforms = _TYPICAL_PLATFORMS, arches = _TYPICAL_ARCHES, sha256_values = {}):
+    """Create a `list` of asset `struct` values.
+
+    Args:
+        version: The buildtools version string.
+        names: Optional. A `list` of tools to include.
+        platforms: Optional. A `list` of platforms to include.
+        arches: Optional. A `list` of arches to include.
+
+    Returns:
+        A `list` of buildtools assets.
+    """
+    if version == None:
+        fail("Expected a version.")
+    if names == None or names == []:
+        fail("Expected a non-empty list for names.")
+    if platforms == None or platforms == []:
+        fail("Expected a non-empty list for platforms.")
+    if arches == None or arches == []:
+        fail("Expected a non-empty list for arches.")
+    if sha256_values == None:
+        sha256_values = {}
+
+    assets = []
+    for name in names:
+        for platform in platforms:
+            for arch in arches:
+                uniq_name = _create_unique_name(
+                    name = name,
+                    platform = platform,
+                    arch = arch,
+                )
+                assets.append(_create_asset(
+                    name = name,
+                    platform = platform,
+                    arch = arch,
+                    version = version,
+                    sha256 = sha256_values.get(uniq_name),
+                ))
+    return assets
+
+_DEFAULT_ASSETS = _create_assets(version = "4.2.3", sha256_values = {
+    "buildifier_darwin_amd64": "954ec397089344b1564e45dc095e9331e121eb0f20e72032fcc8e94de78e5663",
+    "buildozer_darwin_amd64": "edcabae1d97bdc42559d7d1d65dfe7f8970db8d95d4bc9e7bf6656a9f2fb5592",
```

Comment:
i think some of these should be x86_64?

Reply:
Per my last comment, `buildtools` only seems to release `amd64` and `arm64`. https://github.com/bazelbuild/buildtools/releases/tag/4.2.5